File: Java/Linked List/Singly Circular Linked List/SinglyCL02.java
/*
=====================================================================

Name            : Singly Linked List - Three Node Creation
Program Number  : 002
Description     : This program demonstrates creation of three
                  nodes and linking them together to form a
                  simple Singly Linked List.

Concepts Used   : Node Creation, Object References,
                  Link Establishment, Singly Linked List

Input           : None

Output          : Data of all nodes

Date            : July 2026
Language        : Java
Author          : Atharv Tushar Bhosale

=====================================================================
*/

class node
{
    public int data;
    public node next;
}

class SinglyCL02
{
    public static void main(String A[])
    {
        node first = new node();
        node second = new node();
        node third = new node();

        first.data = 11;
        second.data = 21;
        third.data = 51;

        first.next = second;
        second.next = third;
        third.next = null;

        System.out.println(first.data);
        System.out.println(second.data);
        System.out.println(third.data);
    }
}

/*
=====================================================================

Sample Output :

11
21
51

=====================================================================

Observation :

- Three nodes are created.
- Nodes are connected using next reference.
- Last node points to NULL.

=====================================================================
*/

File: C/Linked List/Singly Circular Linked List/SinglyCL02.c
/*
=====================================================================

Name            : Singly Circular Linked List - InsertFirst Operation
Program Number  : 002

Description     : This program demonstrates insertion of
                  nodes at the beginning of a Singly
                  Circular Linked List.

                  The program creates nodes dynamically
                  and maintains the circular link between
                  first and last nodes.

Concepts Used   : Structure,
                  Dynamic Memory Allocation,
                  Pointer Manipulation,
                  Circular Linked List,
                  InsertFirst Operation

Input           : Node data

Output          : Singly Circular Linked List after
                  InsertFirst operations

Date            : July 2026
Language        : C
Author          : Atharv Tushar Bhosale

=====================================================================
*/

// Singly Circular Linked List

#include<stdio.h>
#include<stdlib.h>

#pragma pack(1)

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Display(PNODE first, PNODE last)
{

}

int Count(PNODE first, PNODE last)
{
    return 0;
}

void InsertFirst(PPNODE first, PPNODE last, int iNo)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn -> data = iNo;
    newn -> next = NULL;

    if(*first == NULL && *last == NULL)
    {
        *first = newn;
        *last = newn;
    }
    else
    {
        newn -> next = *first;
        *first = newn;
    }

    (*last) -> next = *first;
}

void InsertLast(PPNODE first, PPNODE last, int iNo)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn -> data = iNo;
    newn -> next = NULL;

    if(*first == NULL && *last == NULL)
    {
        *first = newn;
        *last = newn;
    }
    else
    {

    }

    (*last) -> next = *first;
}

void InsertAtPos(PPNODE first, PPNODE last, int iPos, int iNo)
{}

void DeleteFirst(PPNODE first, PPNODE last)
{}

void DeleteLast(PPNODE first, PPNODE last)
{}

void DeleteAtPos(PPNODE first, PPNODE last, int iPos)
{}

int main()
{
    PNODE head = NULL;
    PNODE tail = NULL;

    InsertFirst(&head,&tail,51);
    InsertFirst(&head,&tail,21);
    InsertFirst(&head,&tail,11);

    Display(head,tail);

    return 0;
}

/*
=====================================================================

Program Summary :

-> Created Singly Circular Linked List nodes
-> Implemented InsertFirst operation
-> Maintained circular connection
-> Updated first and last pointers correctly
-> Prepared foundation for further operations

=====================================================================
*/
